internal/filter/expr/matcher.go

```diff
@@ -63,3 +63,31 @@ func Or[K any](matchers ...BoolExpr[K]) BoolExpr[K] {
 		return orMatcher[K]{matchers: matchers}
 	}
 }
+
+type andMatcher[K any] struct {
+	matchers []BoolExpr[K]
+}
+
+func (am andMatcher[K]) Eval(ctx context.Context, tCtx K) (bool, error) {
+	for i := range am.matchers {
+		ret, err := am.matchers[i].Eval(ctx, tCtx)
+		if err != nil {
+			return false, err
+		}
+		if !ret {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func And[K any](matchers ...BoolExpr[K]) BoolExpr[K] {
```

Comment:
Could I ask you to make a comment on the exported methods?

Reply:
I documented the Eval method inside the interface, the types, and the constructor methods